File: csharp/StrangeCharactersRefactoring/TestCharacters/CharacterDataParserTest.cs
using Characters;
using NUnit.Framework;

namespace TestCharacters;

public class CharacterDataParserTest
{
    [TestCase]
    public void FindCharacterByPath()
    {
        CharacterDataParser.Probably_InitializeFromFile_AndStuff(null);
        var character = CharacterDataParser.EvaluatePath("/Jim/Eleven");
        Assert.AreEqual("Eleven", character?.FirstName);
    }
    
    [TestCase]
    public void FindCharacterByEmptyPath()
    {
        CharacterDataParser.Probably_InitializeFromFile_AndStuff(null);
        var character = CharacterDataParser.EvaluatePath("");
        Assert.Null(character);
    }

    [TestCase]
    public void FindCharacterByPathWithFamilyName()
    {
        CharacterDataParser.Probably_InitializeFromFile_AndStuff(null);
        var character = CharacterDataParser.EvaluatePath("/Wheeler:Karen/Wheeler:Nancy");
        Assert.AreEqual("Nancy", character?.FirstName);
    }
    
    [TestCase]
    public void FindNemesisByPath()
    {
        CharacterDataParser.Probably_InitializeFromFile_AndStuff(null);
        var character = CharacterDataParser.EvaluatePath("/Joyce/Will{Nemesis}");
        Assert.AreEqual("Mindflayer", character?.FirstName);
    }
    
    [TestCase]
    public void FindNemesisByPathAndFamilyName()
    {
        CharacterDataParser.Probably_InitializeFromFile_AndStuff(null);
        var character = CharacterDataParser.EvaluatePath("/Wheeler:Karen/Wheeler:Nancy{Nemesis}");
        Assert.AreEqual(null, character);
    }
    
    [TestCase]
    public void FindNothingByPathAndFamilyName()
    {
        CharacterDataParser.Probably_InitializeFromFile_AndStuff(null);
        var character = CharacterDataParser.EvaluatePath("/Wheeler:Karen/Wheeler:George");
        Assert.AreEqual(null, character);
    }
}

File: php/tests/CharacterDataParserTest.php
<?php

declare(strict_types=1);

use PHPUnit\Framework\Attributes\Test;
use PHPUnit\Framework\TestCase;
use StrangeCharacters\CharacterDataParser;

class CharacterDataParserTest extends TestCase
{
    #[Test]
    public function findCharacterByPath(): void
    {
        CharacterDataParser::probably_InitializeFromFile_AndStuff(null);
        self::assertEquals("Eleven", CharacterDataParser::evaluatePath("/Jim/Eleven")->firstName);
    }

    #[Test]
    public function findCharacterByEmptyPath(): void
    {
        CharacterDataParser::probably_InitializeFromFile_AndStuff(null);
        self::assertNull(CharacterDataParser::evaluatePath(""));
    }

    #[Test]
    public function FindCharacterByPathWithFamilyName(): void
    {
        CharacterDataParser::probably_InitializeFromFile_AndStuff(null);
        self::assertEquals("Nancy", CharacterDataParser::evaluatePath("/Wheeler:Karen/Wheeler:Nancy")->firstName);
    }

    #[Test]
    public function FindNemesisByPath(): void
    {
        CharacterDataParser::probably_InitializeFromFile_AndStuff(null);
        self::assertEquals("Mindflayer", CharacterDataParser::evaluatePath("/Joyce/Will{Nemesis}")->firstName);
    }

    #[Test]
    public function FindNemesisByPathAndFamilyName(): void
    {
        CharacterDataParser::probably_InitializeFromFile_AndStuff(null);
        self::assertNull(CharacterDataParser::evaluatePath("/Wheeler:Karen/Wheeler:Nancy{Nemesis}"));
    }

    #[Test]
    public function FindNothingByPathAndFamilyName(): void
    {
        CharacterDataParser::probably_InitializeFromFile_AndStuff(null);
        self::assertNull(CharacterDataParser::evaluatePath("/Wheeler:Karen/Wheeler:George"));
    }
}
